Arm/Insts/Common.lean

```diff
@@ -368,4 +368,89 @@ def elem_set (vector : BitVec n) (e : Nat) (size : Nat)
   have h : hi - lo + 1 = size := by simp only [hi, lo]; omega
   BitVec.partInstall hi lo (h ▸ value) vector
 
+----------------------------------------------------------------------
+
+structure ShiftRightInfo where
```

Comment:
Curious why you defined `ShiftRightInfo` and `ShiftLeftInfo` instead of just one structure, `ShiftInfo`. The fields in `ShiftRightInfo` that aren't relevant to the left-shift can be optional.

Reply:
Great idea.